rust/connlib/snownet/src/node.rs

```diff
@@ -1350,22 +1351,16 @@ where
     fn insert_established(
         &mut self,
         id: TId,
-        index: u32,
+        index: Index,
         connection: Connection<RId>,
     ) -> Option<Connection<RId>> {
         let existing = self.established.insert(id, connection);
 
-        debug_assert_eq!(
-            index >> 24,
-            0,
-            "The 8 most-significant bits should always be zero."
-        );
-
         // Remove previous mappings for connection.
         self.established_by_wireguard_session_index
             .retain(|_, c| c != &id);
         self.established_by_wireguard_session_index
-            .insert(index, id);
+            .insert(index.global(), id);
```

Comment:
Because the index is now properly typed, we can perform the insert and lookup of peers based on their global index without having to do any bit-shifting.